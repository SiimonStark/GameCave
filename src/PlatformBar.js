import React, { Component } from 'react';
import CheckBox from './CheckBox.js';

export default class PlatformBar extends Component {
  constructor(props) {
    super(props);
    this.state = {showPlatforms: false, mouseIn: false};
  }
  togglePlatforms = () => {
    this.setState({showPlatforms: this.state.showPlatforms ? false : true});
  }
  toggleMouse = () => {
    this.setState({ mouseIn: this.state.mouseIn ? false : true });
  }
  focusLost = () => {
    this.setState({ showPlatforms: this.state.mouseIn ? true : false })
  }
  render() {
    let hidePlatforms = this.state.showPlatforms ? "platform-check" :  "platform-check hide";
    let buttonSelector = this.props.platformsToFilter.length ? "button current" : "button";
    let platforms = this.props.platforms.sort();
    return (
      <div onBlur={this.focusLost} onMouseEnter={this.toggleMouse} onMouseLeave={this.toggleMouse} className="console-bar search-section">
          <form onBlur={this.focusLost} tabIndex="0" className={hidePlatforms}>
            {
              platforms.map((val, i) => {
                return <CheckBox name={val}
                  updateChecks= {this.props.updateChecks}
                  class="platformsToFilter check-box"
                  key={i} />
              })
            }
          </form>
        <button onClick={this.togglePlatforms} className={buttonSelector}>PLATFORM</button>
      </div>
    )
  }
}
