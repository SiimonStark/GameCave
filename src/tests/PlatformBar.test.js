import React from 'react';
import ReactDOM from 'react-dom';
import PlatformBar from '../PlatformBar';
import { shallow } from 'enzyme';

// const updateStateMock = jest.fn();

const plats = ['PS4', 'PS3', 'XBOX']
const toFilterMock = ['PS4', 'PS3']
describe('PlatformBar', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <PlatformBar 
        platformsToFilter={toFilterMock}
        platforms={plats}
        />
    );
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    expect(wrapper.state()).toEqual({
      showPlatforms: false, mouseIn: false
    });
  })

  it('should toggle mouseIn on mouse leave', () => {
    expect(wrapper.state('mouseIn')).toEqual(false);
    wrapper.find('.console-bar').simulate('mouseleave');
    expect(wrapper.state('mouseIn')).toEqual(true);
  })

  it('should toggle mouseIn on mouse enter', () => {
    expect(wrapper.state('mouseIn')).toEqual(false);
    wrapper.find('.console-bar').simulate('mouseleave');
    expect(wrapper.state('mouseIn')).toEqual(true);
    wrapper.find('.console-bar').simulate('mouseenter');
    expect(wrapper.state('mouseIn')).toEqual(false);
  })

  it('should toggle showPlatforms when button is clicked', () => {
    expect(wrapper.state('showPlatforms')).toEqual(false);
    wrapper.find('.button').simulate('click');
    expect(wrapper.state('showPlatforms')).toEqual(true);
  })

  it('should change showPlatforms on blur', () => {
    wrapper.find('.button').simulate('click');
    expect(wrapper.state('showPlatforms')).toEqual(true);
    wrapper.find('.console-bar').simulate('blur');
    expect(wrapper.state('showPlatforms')).toEqual(false);
  })

  // it('should invoke updateState upon input change', () => {
  //   wrapper.find('form').simulate('change');
  //   expect(updateStateMock).toBeCalled();
  // })
})