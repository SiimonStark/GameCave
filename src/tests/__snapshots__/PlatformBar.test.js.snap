// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformBar should match snapshot 1`] = `
<div
  className="console-bar search-section"
  onBlur={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <form
    className="platform-check hide"
    onBlur={[Function]}
    tabIndex="0"
  >
    <CheckBox
      class="platformsToFilter check-box"
      key="0"
      name="PS3"
    />
    <CheckBox
      class="platformsToFilter check-box"
      key="1"
      name="PS4"
    />
    <CheckBox
      class="platformsToFilter check-box"
      key="2"
      name="XBOX"
    />
  </form>
  <button
    className="button current"
    onClick={[Function]}
  >
    PLATFORM
  </button>
</div>
`;
